Fixed the issue where the record always links to the latest version

fallen-axe-shay opened the pull request.

fallen-axe-shay committed:
Fixed the issue where the record always links to the latest version

carboxylman commented:
The problem with this is that we don't want users to copy full-version URLs and link to them in the default case.  We only want to show them a version-specific URL when they are viewing/editing something other than the current published version.

This is a simple conditional -- so I would replace this inline URL with a generator function that takes as input an artifact dict, and does something like

```
getArtifactLink(artifact) {
if (artifact.artifact_group !== 'undefined' && artifact.artifact_group.publication !== 'undefined' && artifact.artifact_group.publication.artifact.id == artifact.id)
  return `/artifact/${artifact.artifact_group_id}`"
else
  return `/artifact/${artifact.artifact_group_id}/${artifact.id}`"
}
```

fallen-axe-shay committed:
Addressed PR Comments

fallen-axe-shay committed:
Added Contributor icon in my artifacts

fallen-axe-shay committed:
Added Draft chip functionality

fallen-axe-shay commented:
Addressed the issue where duplicates were found in "My Artifacts". I've also added chips to differentiate between the user roles for a specific artifact (draft, contributor, owner).

I've also attached 2 screenshots - one in My Artifacts, and one in Artifact Search.

![My Artifacts](https://user-images.githubusercontent.com/89501492/195929523-8760b18d-2f58-418c-aaaa-47f6706c04f6.png)

![Artifact Search](https://user-images.githubusercontent.com/89501492/195929714-b0ae8626-17a7-414c-ba73-5d065bf6fe1e.png)
